packages/react-storage/src/components/StorageBrowser/Views/LocationsListView/LocationsListView.tsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { useDataState } from '@aws-amplify/ui-react-core';
+import { listLocationsAction } from '../../context/actions';
+import { Controls } from '../Controls';
+import { CommonControl, ViewComponent } from '../types';
+import { StorageBrowserElements } from '../../context/elements';
+import { CLASS_BASE } from '../constants';
+
+const { Divider, Message, Navigate, Paginate, Refresh, Search, Table, Title } =
+  Controls;
+
+interface LocationsListViewControls<
+  T extends StorageBrowserElements = StorageBrowserElements,
+  // exlcude `Toggle` from `Search` for Locations List
+> extends Exclude<
+    Pick<Controls<T>, CommonControl | 'Paginate' | 'Refresh' | 'Search'>,
+    Controls<T>['Search']['Toggle']
+  > {
+  (): React.JSX.Element;
+}
+
+export interface LocationsListView<
+  T extends StorageBrowserElements = StorageBrowserElements,
+> extends ViewComponent<T, LocationsListViewControls<T>> {}
+
+const LocationsListViewControls: LocationsListViewControls = () => (
+  <div className={`${CLASS_BASE}__header`}>
+    <Navigate />
+    <div className={`${CLASS_BASE}__header__primary`}>
+      <Title />
+      <div className={`${CLASS_BASE}__header__primary__actions`}>
+        <Refresh />
+      </div>
+    </div>
+    <div className={`${CLASS_BASE}-header__secondary`}>
```

Comment:
nit: Should this use the `__` after CLASS_BASE, like the other classes you have above?

Reply:
@hbuchel thoughts? @timngyn  this code was migrated and updated minimally